tests/test_extractors.py

```diff
@@ -0,0 +1,338 @@
+        cls.resource_path = os.path.join(os.environ['TESTS_RESOURCES_DIR'], 'extractors/helper')
+
+    def test_invert_yang_modules_dict(self):
+        test_dict = {'test_key': ['test_yang_model_file1', 'test_yang_model_file2', 'test_yang_model_file3']}
+        yang_models = test_dict['test_key']
+        result = helper.invert_yang_modules_dict(test_dict, debug_level=1)
+        self.assertEqual(len(result), len(yang_models))
+        for yang_model in yang_models:
+            self.assertEqual(result[yang_model], 'test_key')
+
+    def test_remove_invalid_files(self):
+        directory = os.path.join(self.resource_path, 'remove_invalid_files')
+        yang_dict = {}
+        backup_dir = os.path.join(self.resource_path, 'backup_dir')
+        os.makedirs(backup_dir, exist_ok=True)
+        for filename in os.listdir(directory):
+            if os.path.isfile(filepath := os.path.join(directory, filename)):
+                yang_dict[filename] = 'test value'
+                shutil.copy2(filepath, backup_dir)
+        try:
+            helper.remove_invalid_files(directory, yang_dict)
+            self.assertEqual(len(os.listdir(directory)), 1)
+            self.assertEqual(len(yang_dict), 1)
+            self.assertIn('valid_filename@1999-12-12.yang', yang_dict)
+        finally:
+            for filename in os.listdir(backup_dir):
+                if os.path.isfile(filepath := os.path.join(backup_dir, filename)):
+                    shutil.copy2(filepath, directory)
+            shutil.rmtree(backup_dir)
+
+
+class TestDraftExtractor(unittest.TestCase):
+    resource_path: str
+    drafts_path: str
+    draft_extraction_paths: str
+    draft_extractor_paths: draft_extractor.DraftExtractorPaths
+    problematic_drafts_dir: str
+
+    @classmethod
+    def setUpClass(cls):
+        cls.resource_path = os.path.join(os.environ['TESTS_RESOURCES_DIR'], 'extractors/draft_extractor')
+        cls.drafts_path = os.path.join(cls.resource_path, 'drafts')
+        cls.draft_extraction_paths = os.path.join(cls.resource_path, 'draft_extraction_paths')
+        cls.draft_extractor_paths = draft_extractor.DraftExtractorPaths(
+            draft_path=cls.drafts_path,
+            yang_path=os.path.join(cls.draft_extraction_paths, 'yang'),
+            draft_elements_path=os.path.join(cls.draft_extraction_paths, 'draft_elements'),
+            draft_path_strict=os.path.join(cls.draft_extraction_paths, 'draft_strict'),
+            all_yang_example_path=os.path.join(cls.draft_extraction_paths, 'all_yang_example'),
+            draft_path_only_example=os.path.join(cls.draft_extraction_paths, 'only_examples_drafts'),
+            all_yang_path=os.path.join(cls.draft_extraction_paths, 'all_yang'),
+            draft_path_no_strict=os.path.join(cls.draft_extraction_paths, 'draft_no_strict'),
+            code_snippets_dir=os.path.join(cls.draft_extraction_paths, 'code_snippets'),
+        )
+        for path in cls.draft_extractor_paths.__dict__.values():
```

Comment:
Does this only actually give you strings? Doesn't the dict also contain methods and other things? I thik `dataclasses.asdict` is supposed to provide this functionality, but if this works as well that's interesting.

Reply:
Yes it works perfectly fine, I was also surprised by this
![image](https://github.com/YangCatalog/module-compilation/assets/60445869/042153a2-1391-4907-9cd5-9fd2ca7792f4)
but I rewrote it with `asdict()` because it's a proper solution, thanks